spec/features/result_filters/location_spec.rb

```diff
@@ -5,10 +5,35 @@
   let(:provider_page) { PageObjects::Page::ResultFilters::ProviderPage.new }
   let(:results_page) { PageObjects::Page::Results.new }
   let(:query_params) { {} }
+  let(:default_url) do
+    "http://localhost:3001/api/v3/recruitment_cycles/2020/courses"
+  end
+
+  let(:base_parameters) do
+    {
+        "filter[vacancies]" => "true",
+        "filter[qualifications]" => "QtsOnly,PgdePgceWithQts,Other",
+        "include" => "provider",
+        "page[page]" => 1,
+        "page[per_page]" => 10,
+    }
+  end
 
   before do
     stub_geocoder
-    stub_results_page_request
+
+    stub_api_v3_resource(
```

Comment:
I've just been having a look through this method. I think it bypasses the normal deserialization of the json and uses tests specific stuff. If that is what's happening I think for feature specs where we're trying to test the full stack we should stub the network response with webmock. That way we are testing the code in the model as well. We could use `stub_api_v3_request` for that I think but it takes a hash so we'd have the overhead of getting an example response in JSON and converting it. That's ok for small bits of data but in these specs the responses are quite big.

Reply:
👍